plugins/event_source/eventstream.py

```diff
@@ -0,0 +1,293 @@
+            if line:
+                # Load the event as a JSON object
+                json_event = json.loads(line)
+                event_type = json_event["metadata"]["eventType"]
+                self.offset = json_event["metadata"]["offset"]
+                # If the event is valid, yield it
+                if self.is_valid_event(event_type, exclude_event_types):
+                    # If INFO or DEBUG, update the event_type_count dict and yield it
+                    if logger.level <= logging.INFO:
+                        self.event_type_count[event_type] = self.event_type_count.get(event_type, 0) + 1
+                        yield dict(falcon=json_event, eventTypeCount=self.event_type_count)
+                    else:
+                        yield dict(falcon=json_event)
+            # If the token has expired, refresh it and reopen the stream
+            if self.token_expired():
+                await self.refresh()
+                continue
+
+    def is_valid_event(self, event_type: str, exclude_event_types: List[str]) -> bool:
+        """
+        This function checks if a given event type is valid or not.
+
+        Parameters:
+            event_type (str): The type of the event to be checked.
+            exclude_event_types (List[str]): A list of event types to be excluded.
+
+        Returns:
+            bool: Returns False if the event_type is in the exclude_event_types list, otherwise returns True.
+        """
+        if event_type in exclude_event_types:
+            return False
+        return True
+
+
+async def main(queue: asyncio.Queue, args: Dict[str, Any]) -> None:
+    """
+    Main function for the eventstream event_source plugin
+
+    Args:
+        queue (asyncio.Queue): The queue to send events to
+        args (Dict[str, Any]): The event_source arguments
+
+    Raises:
+        ValueError: If a argument is invalid
+    """
+    falcon_client_id: str = str(args.get("falcon_client_id"))
+    falcon_client_secret: str = str(args.get("falcon_client_secret"))
+    falcon_cloud: str = str(args.get("falcon_cloud", "us-1"))
+    stream_name: str = str(args.get("stream_name", "eda")).lower()
+    offset: int = int(args.get("offset", 0))
+    delay: float = float(args.get("delay", 0))
+    include_event_types: List[str] = list(args.get("include_event_types", []))
+    exclude_event_types: List[str] = list(args.get("exclude_event_types", []))
+
+    if falcon_cloud not in REGIONS:
+        raise ValueError(f"Invalid falcon_cloud: {falcon_cloud}, must be one of {list(REGIONS.keys())}")
+
+    falcon: APIHarness = APIHarness(client_id=falcon_client_id,
+                                    client_secret=falcon_client_secret,
+                                    base_url=REGIONS[falcon_cloud],
+                                    ssl_verify=True
+                                    )
+
+    loop = asyncio.get_running_loop()
+    authenticated = await loop.run_in_executor(None, falcon.authenticate)
```

Comment:
@carlosmmatos Write async method to handle authentication and event stream api calls. Can't use falconpy